lib/systems/examples.nix

```
@@ -221,6 +221,32 @@ rec {
     libc = "newlib";
   };
 
+  #
+  # UEFI
+  #
+
+  # See: https://github.com/rust-lang/rust/pull/56769/files#r241916113
+  i686-uefi = {
+    system = "x86_64-windows";
```

This causes `platform.config` to be `i686-unknown-uefi` which is rejected by `config.sub`:

```
$ /nix/store/ayj17jhbiqc1k043ld3nnb3xqs0p22sl-gnu-config-2023-01-21/config.sub i686-unknown-uefi
Invalid configuration `i686-unknown-uefi': OS `uefi' not recognized
```

so please:

```suggestion
    config = "x86_64-w64-mingw32";
```

This is the `gnu-config` triple for this platform.   Please use it.  See https://github.com/NixOS/nixpkgs/pull/226145#pullrequestreview-1403683064

---

> causes `platform.config` to be `i686-unknown-uefi` which is rejected by `config.sub`:
> 
> ```
> $ /nix/store/ayj17jhbiqc1k043ld3nnb3xqs0p22sl-gnu-config-2023-01-21/config.sub i686-unknown-uefi
> Invalid configuration `i686-unknown-uefi': OS `uefi' not recognized
> ```

No it doesn't:

```
nix-repl> (lib.systems.elaborate "i686-windows").config
"i686-pc-windows-msvc"
```

config.sub won't recognise this either, but that's because GNU doesn't support the MSVC ABI, so there's no correct triple to use here.  This is correct for LLVM, which is what we actually use with these system definitions.

Are you just copying these comments from the other PR without checking whether they apply to this one?  It seems like you're arguing against things that this PR isn't trying to do…